csrc/scheduler/matmul_utils.cpp

```diff
@@ -535,7 +535,23 @@ std::shared_ptr<MatmulParams> getMatmulHeuristics(
   // Set kernel index mode
   params->cparams.index_type = getIndexType(problem_shape.value());
 
-  if (isDebugDumpEnabled(DebugDumpOption::MatmulChecks)) {
+  // Set split k using input tensor size M,N,K
+  const auto& MNK = problem_shape.value();
+  const int sm_count = device_prop->multiProcessorCount;
+  const auto& cta_tile = params->tile_sizes.cta_tile;
+  const int num_blocks =
+      ceilDiv(MNK[0], cta_tile.m) * ceilDiv(MNK[1], cta_tile.n);
+  const int iter_k = ceilDiv(MNK[2], cta_tile.k);
+  if (num_blocks < sm_count && iter_k > 32) {
```

Comment:
I have two suggestions/questions regarding the split-k heuristic:

- Shouldn't it take into account the # of warps/cta too? For example we have single-warp CTA tiles. Even if num_blocks == sm_count, we're not filling the GPU at all. In that case we would ideally split by 4 to have at least 4 warps/sm (to use all of the tensors cores). Most of the GEMM kernels are about 240reg/thread. So we can assume a max of 8 resident warps/SM.
- Shouldn't it also take into account the GPU/Waves utilization for "tail effects"? For ex with 256 threads CTA tiles, `sm_count =100`, `num_blocks =150` we will also want to use split-k.

Reply:
Thanks for these suggestions. Yes we definitely need to consider these factors. This PR focus on the functionality so the current "heuristics" for split-k is very simple. The performance will be addressed later, probabally combined with other split-k versions.

Reply:
Sure! I also need to check the latencies introduced by the various split-k factors. Maybe you could add a test/case with `M=128, N = 128*num_sm, K=8192` and check several k values?

Reply:
will do after the "split-k heuristics" is updated.